### DISCUSSION TITLE
Hydra - AWS Aurora MySQL - Passwordless Authentication

### DISCUSSION THREAD

**Discussion by jasonn514** [ory/.github#8] [Q&A] [answered]
Hello,

I'm new to hydra and looking to deploy it in our environment.  We are working on deploying hydra using Amazon AWS Aurora MySQL. The issue we are having is that we are not permitted to use username / password to access the MySQL DB. 

We tried using Amazon STS tokens (which replace the password but expires at a regular interval) but Hydra seems to have issues with the format of the token even if we url encode it

Sample "STS Token / Password":

`host.subdomain.us-east-1.ramazonaws.com:6140/?Action=connect&DBUser=DBAppAdmin&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Expires=900&X-Amz-Date=20220510T184954Z&X-Amz-SignedHeaders=host&X-Amz-Security-Token=FwoGZXIvYXdzEFQH%2FZBiq1IeGXMMxG1eTBvQvYb2X6LrHBvJcds5lHVxUAEJi&X-Amz-Credential=ASIAWU%2F20220510%2Fus-east-1%2Frds-db%2Faws4_request&X-Amz-Signature=21e95b166dfd4a4b45561a5ad45322c949f9aeee861555a3db `

Are there alternative authentication methods that do not rely on username / password like Mutual TLS that we can use to connect to MySQL (or other database)?

Thank you,
Jason

**Comment by aeneasr** [comment, marked as answer]
Hi @jasonn514 we use the go-mysql driver under the hood and support all the options they support: https://github.com/go-sql-driver/mysql#dsn-data-source-name
